src/ajax.js

```diff
@@ -1,56 +1,55 @@
 import {parse as parseURL, format as formatURL} from './url';
 
-/**
- * Simple cross-browser ajax request function
- * https://gist.github.com/Xeoncross/7663273
+const XHR_DONE = 4;
 
- * IE 5.5+, Firefox, Opera, Chrome, Safari XHR object
+/**
+ * Simple IE9+ and cross-browser ajax request function
  *
  * @param url string url
  * @param callback object callback
  * @param data mixed data
  * @param options object
  */
 
-export const ajax = function ajax(url, callback, data, options = {}) {
-  let x;
-
-  try {
-    if (window.XMLHttpRequest) {
-      x = new window.XMLHttpRequest('MSXML2.XMLHTTP.3.0');
-    }
+export function ajax(url, callback, data, options = {}) {
 
-    if (window.ActiveXObject) {
-      x = new window.ActiveXObject('MSXML2.XMLHTTP.3.0');
-    }
+  let x,
+      method = options.method || (data ? 'POST' : 'GET'),
+      // For IE9 support use XDomainRequest instead of XMLHttpRequest.
+      useXDomainRequest = window.XDomainRequest &&
+        (!window.XMLHttpRequest || new window.XMLHttpRequest().responseType === undefined);
 
-    const method = options.method || (data ? 'POST' : 'GET');
+  if (useXDomainRequest) {
+    x = new window.XDomainRequest();
+    x.onload = handler;
```

Comment:
Can you add the following here: 
```
    x.onerror = function () {
         utils.logMessage('xhr onerror');
     };
     x.ontimeout = function () {
         utils.logMessage('xhr timeout');
     };
     x.onprogress = function() {
         utils.logMessage('xhr onprogress');
     };
```

When I was testing this in IE 9, the POST would randomly abort. After adding all the handlers, it seems to succeed all the time. See this link: http://stackoverflow.com/questions/15786966/xdomainrequest-aborts-post-on-ie-9